httpu/httpu.go

```diff
@@ -26,6 +27,25 @@ type ClientInterface interface {
 	) ([]*http.Response, error)
 }
 
+// ClientInterfaceCtx is the equivalent of ClientInterface, except with methods
+// taking a context.Context parameter.
+type ClientInterfaceCtx interface {
```

Comment:
I created a separate interface for this, so as not to break people that implement the `ClientInterface` interface already. I can put this in the main interface, if you'd prefer; it makes some of the code below more ugly to have this as a separate interface.